packages/create-cloudflare/src/helpers/poll.ts

```diff
@@ -1,7 +1,8 @@
-import { Resolver } from "node:dns/promises";
+import dns2 from "dns2";
 import { request } from "undici";
 import { blue, brandColor, dim } from "./colors";
 import { spinner } from "./interactive";
+import type { DnsAnswer, DnsResponse } from "types";
 
```

Comment:
could you please add a file specific comment block explaining at high level how the polling works?

we've made it a bit non-trivial to work around all the negative dns cache hits, so for future maintenance it would help if we captured how and why it works the way it works while we still remember :)